src/Microsoft.AspNet.Hosting/HostingEngine.cs

```diff
@@ -47,14 +50,21 @@ public IDisposable Start(HostingContext context)
             InitalizeServerFactory(context);
             EnsureApplicationDelegate(context);
 
+            var loggerFactory = context.ApplicationServices.GetRequiredService<ILoggerFactory>();
+            var logger = loggerFactory.CreateLogger<HostingEngine>();
             var contextFactory = context.ApplicationServices.GetRequiredService<IHttpContextFactory>();
             var contextAccessor = context.ApplicationServices.GetRequiredService<IHttpContextAccessor>();
             var server = context.ServerFactory.Start(context.Server,
-                features =>
+                async features =>
                 {
-                    var httpContext = contextFactory.CreateHttpContext(features);
-                    contextAccessor.HttpContext = httpContext;
-                    return context.ApplicationDelegate(httpContext);
+                    var requestIdentifier = GetRequestIdentifier(features);
```

Comment:
Sure, I am not too strong on this as currently its `Guid.Empty` and not useful...a related question: Should we create a default request identifier feature in case one is not provided by the host?(ex: Kestrel)...I deliberately did not create a new Guid over here as this would be specific to logging scope and a user might try to get the identifier feature in their own code for whatever reason and would not see this id...